rust/osm2lanes/src/transform.rs

```diff
@@ -502,34 +512,129 @@ fn osm_separation_type(x: &str) -> Option<BufferType> {
 
 pub fn lanes_to_tags(lanes: &[LaneSpec], _cfg: &Config) -> Result<Tags, LaneSpecError> {
     let mut tags = std::collections::BTreeMap::new();
+    let mut oneway = false;
     tags.insert("highway".to_owned(), "yes".to_owned()); // TODO, what?
+                                                         // Lanes
     {
         let lane_count = lanes
             .iter()
-            .filter(|lane| lane.lane_type == LaneType::Driving)
+            .filter(|lane| {
+                lane.lane_type == LaneType::Driving || lane.lane_type == LaneType::SharedLeftTurn
+            })
             .count();
         tags.insert("lanes".to_owned(), lane_count.to_string());
     }
+    // Oneway
     if lanes
         .iter()
         .filter(|lane| lane.lane_type == LaneType::Driving)
         .all(|lane| lane.direction == Direction::Forward)
     {
+        oneway = true;
         tags.insert("oneway".to_owned(), "yes".to_owned());
     }
-    if lanes.first().unwrap().lane_type == LaneType::Sidewalk
-        && lanes.last().unwrap().lane_type == LaneType::Sidewalk
+    // Pedestrian
     {
-        tags.insert("sidewalk".to_owned(), "both".to_string());
+        match (lanes.first().unwrap().lane_type == LaneType::Sidewalk, lanes.last().unwrap().lane_type == LaneType::Sidewalk)
+        {
+            (false, false) => {}
+            (true, false) => assert!(tags.insert("sidewalk".to_owned(), "left".to_owned())
```

Comment:
Might I suggest adding another helper to `Tags` along the lines of:
```
pub fn insert<K: Into<String>, V: Into<String>>(&mut self, k: K, v: V) {
        self.0.insert(k.into(), v.into());
}
```

And maybe it can also assert the key isn't present yet

Reply:
I need to refactor tags into trait implementations, will make this part of it :)